File: 프로그래머스/0/120880. 특이한 정렬/특이한 정렬.java
import java.util.Arrays;

class Solution {
    public Integer[] solution(int[] numlist, int n) {
        Integer[] copy = new Integer[numlist.length];
        
        for (int i = 0; i < numlist.length; i++) {
            copy[i] = numlist[i];
        }
        
        Arrays.sort(copy, (i1, i2) -> {
            int a1 = Math.abs(i1 - n);
            int a2 = Math.abs(i2 - n);
            
            if (a1 == a2) {
                return i2 - i1;
            }
            return a1 - a2;
        });
        
        return copy;
    }
}

File: 프로그래머스/0/120880. 특이한 정렬/특이한 정렬.js
function solution(numlist, n) {
    return numlist.sort((n1, n2) => {
        const abs1 = Math.abs(n - n1);
        const abs2 = Math.abs(n - n2);
        
        if (abs1 !== abs2) {
            return abs1 - abs2;
        }
        return n2 - n1;
    });
}